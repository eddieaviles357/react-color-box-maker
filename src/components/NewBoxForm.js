import React from 'react';

// this component should render a form that when submitted, creates a new Box. 
// You should be able to specify the Boxâ€™s width, height, and background color. 
// When the form is submitted, clear the input values.

function NewBoxForm(props) {

    const handleSubmit = (evt) => {
        evt.preventDefault();
        console.log('submit')
    }
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor='width'>Width</label>
            <input type="text" id='width' name='width' />
            <label htmlFor='height'>Height</label>
            <input type="text" id='height' name='height' />
            <label htmlFor='color'>Color</label>
            <input type='color' id='color' name='color' />
            <button>Submit</button>
        </form>
    )
}

export default NewBoxForm;