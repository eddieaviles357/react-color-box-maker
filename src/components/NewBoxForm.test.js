import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('should render without crashing', () => {
    render(<NewBoxForm />);
})

it('matches snapshot', () => {
    const {asFragment, getByText} = render(<NewBoxForm />);

    expect(asFragment()).toMatchSnapshot();
});

it('should display text Width and Height and Submit', () => {
    const {getByText} = render(<NewBoxForm />);

    expect(getByText('Width:')).toBeInTheDocument();
    expect(getByText('Height:')).toBeInTheDocument();
    expect(getByText('Submit')).toBeInTheDocument();
})

it('should display Color #232323 black as default', () => {
    const {getByText} = render(<NewBoxForm />);

    expect(getByText('Color: #232323')).toBeInTheDocument();
});

it('should display correct width 100 when range is moved to max range', () => {
    const {getByLabelText, queryByText} = render(<NewBoxForm />);
    const widthInput = getByLabelText('Width:');

    expect( queryByText('Width: 100') ).not.toBeInTheDocument();
    fireEvent.change( widthInput, { target: {value: '100' } } );
    expect( queryByText('Width: 100') ).toBeInTheDocument();
});

it('should display correct height 100 when range is moved to max range', () => {
    const {getByLabelText, queryByText} = render(<NewBoxForm />);
    const heightInput = getByLabelText('Height:');

    expect( queryByText('Height: 100') ).not.toBeInTheDocument();
    fireEvent.change( heightInput, { target: {value: '100' } } );
    expect( queryByText('Height: 100') ).toBeInTheDocument();
});

it('should display correct color when color is choosen', () => {
    const {getByLabelText, queryByText} = render(<NewBoxForm />);
    const colorInput = getByLabelText('Color: #232323');

    expect( queryByText('Color: #006d8f') ).not.toBeInTheDocument();
    fireEvent.change( colorInput, { target: {value: '#006d8f' } } );
    expect( queryByText('Color: #006d8f') ).toBeInTheDocument();
});