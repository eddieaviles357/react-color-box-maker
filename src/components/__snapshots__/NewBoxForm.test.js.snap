// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="App-form"
  >
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      max="100"
      min="25"
      name="width"
      type="range"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      max="100"
      min="25"
      name="height"
      type="range"
      value=""
    />
    <label
      for="color"
    >
      Color: #232323
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value="#232323"
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
